artiq/coredevice/comm_analyzer.py

```diff
@@ -307,6 +307,83 @@ def process_read(self, address, data, read_slack):
         raise NotImplementedError
 
 
+class GenericWishboneHandler(WishboneHandler):
+    """Generic Wishbone Bus Data Handler.
+
+    Useful for testing out-of-tree PHY devices and viewing their bus transactions
+    in a VCD file.
+    """
+
+    comm_buses_and_width = {
+        "out_address": 8,
+        "out_data": 32,
+        "out_stb": 1,
+        "in_data": 14,
+        "in_stb": 1
+    }
+    def __init__(self, vcd_manager, name, channel):
+        """Create the Wishbone Data Handler.
+
+        Args:
+            vcd_manager (VCDManager): VCD Manager to add this device's channels to.
+            name (str): Name of the coredevice that communicates on this channel.
+        """
+        self.channels = {}
+        with vcd_manager.scope("bus_device/{}(chan{})".format(name, channel)):
+            for bus_name, bus_width in self.comm_buses_and_width.items():
+                self.channels[bus_name] = vcd_manager.get_channel(
+                    "{}/{}".format(name, bus_name),
+                    bus_width
+                )
+
+    def set_channel(self, channel_name, value):
+        """Set channel to a specific value.
+
+        Formats the data properly for interpretation by VCD viewer
+        (on a per-channel basis).
+
+        Args:
```

Comment:
This isn't the usual way those are formatted, see e.g.
https://github.com/m-labs/artiq/blob/8c5a50259149cb4963ca23bfd55b8c1d8912a9fe/artiq/coredevice/ad53xx.py#L74

Reply:
I disagree. This is a standard format recognized by Sphinx (using the standard Napoleon extension). https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

The format you cite above I consider much more obtuse and illegible, and doesn't include type information.

Reply:
I for one also prefer the numpy/napoleon style.

Reply:
Can we just stick to `:param foo: bar.` in the ARTIQ code base please? This is not a statement of the relative merits of the two choices, just that uniformity decreases cognitive load. (Also, with type annotations now being a thing in the language proper, there is no need to mangle them into the docstring.)

Reply:
I meant in general. I agree that consistency trumps that preference here.

Reply:
ack. will change.

Reply:
Thanks!

Reply:
Yes, this wasn't an argument about which style is best, but simply an argument for consistency.

If you want to change the format @drewrisinger please submit a separate issue (to discuss it) and then a PR that changes it everywhere in ARTIQ. The same goes for the type annotations that you wanted earlier (though I have some reservations about them).